src/App.tsx

```diff
@@ -15,36 +15,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getVisibleGoods(goods: string [], sortType: string, isReversed: boolean) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case 'alphabetically':
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case 'length':
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visible = getVisibleGoods(goodsFromServer, sortType, isReversed);
+
+  const resetSort = () => {
+    setSortType('');
+    setIsReversed(false);
+  };
+
+  const toggleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabetically' ? '' : 'is-light'}`}
```

Comment:
use [classnames](https://www.npmjs.com/package/classnames) library for setting dynamic classes